pkg/server/container_stop.go

```diff
@@ -141,11 +134,21 @@ func (c *criService) stopContainer(ctx context.Context, container containerstore
 			return errors.Wrapf(err, "failed to stop container %q", id)
 		}
 
-		if err = c.waitContainerStop(ctx, container, timeout); err == nil || errors.Cause(err) == ctx.Err() {
-			// Do not SIGKILL container if the context is cancelled.
-			return err
+		sigTermCtx, sigTermCtxCancel := context.WithTimeout(ctx, timeout)
```

Comment:
I think it is fine to keep the original timeout timer, making it purely context based doesn't make the code simpler.

Reply:
I can do that for the container_stop code if you like it better. When the timer expires we just need to return `errors.Wrapf(context.DeadlineExceeded, "wait container %q stop timeout", container.ID)` and this should work too. Want me to do that?

Reply:
What we lose in that model is the ability for the client context to cause the cancellation

Reply:
I'm fine with using context. I'm just not a fan of having a `cancel` not in `defer`, which may cause trouble when we move code around in the future. :P

Reply:
The call is idempotent. I suppose we could defer it and cancel it early. Want me to send a subsequent update?

Reply:
Lol: https://github.com/containerd/cri/pull/1207